tests/python_tests/helpers/pack.py

```diff
@@ -29,7 +29,7 @@ def bfloat16_to_bytes(number):
     for i in range(0, len(torch_tensor), 2):
         half1 = bfloat16_to_bytes(torch_tensor[i])
         half2 = bfloat16_to_bytes(torch_tensor[i + 1])
-        packed_bytes.extend([half1[0:2][::-1], half2[0:2][::-1]][::-1])
+        packed_bytes.extend([half1[0:2][::-1], half2[0:2][::-1]])
```

Comment:
Although this change fixes matmul test cases and is compatible with other tests in the infrastructure, we are currently only running tests where all data in pipeline is the same format. Bugs were previously found in `pack.py` and `unpack.py` when combining formats that were not the same. This change may override and reverse those bug patchups causing the same tests to start failing when we run them with various format combinations. This fix here is a patch up currently since we haven't found the universal fix for all tests across format combinations. Until we do a full format sweep I would suggest keeping this here but adding a flag that checks if input and output formats are the same, to avoid reversing order of bytes that works for differing format combinations.
```suggestion
        if formats.unpack_A_src == formats.pack_dst:
               packed_bytes.extend([half1[0:2][::-1], half2[0:2][::-1]])
```
  make sure to update function parameters 
`def pack_bfp16(torch_tensor, formats: FormatConfig):`

Reply:
Can we run multiple combinations to see if this is really a fix or not ?